src/metpy/calc/indices.py

```diff
@@ -126,20 +126,17 @@ def mean_pressure_weighted(pressure, *args, height=None, bottom=None, depth=None
        Renamed ``heights`` parameter to ``height``
 
     """
-    ret = []  # Returned variable means in layer
     layer_arg = get_layer(pressure, *args, height=height,
                           bottom=bottom, depth=depth)
-    layer_p = layer_arg[0]
-    layer_arg = layer_arg[1:]
+    layer_p = layer_arg[0]  # Pressure profile
+    layer_arg = layer_arg[1:]  # All other profiles provided to args
+
     # Taking the integral of the weights (pressure) to feed into the weighting
     # function. Said integral works out to this function:
     pres_int = 0.5 * (layer_p[-1]**2 - layer_p[0]**2)
-    for i, _datavar in enumerate(args):
-        arg_mean = np.trapz((layer_arg[i] * layer_p),
-                            x=layer_p) / pres_int
-        ret.append(arg_mean)
 
-    return ret
+    # Perform integration for each layer given to args
+    return [np.trapz((layer_arg[i] * layer_p), x=layer_p) / pres_int for i in range(len(args))]
```

Comment:
Almost nailed what I was thinking. :wink: Just a bit more Pythonic here:
```suggestion
    return [np.trapz(arg * layer_p, x=layer_p) / pres_int for arg in args]
```